go/saltpack/decrypt.go

```diff
@@ -30,6 +29,17 @@ type decryptStream struct {
 	nonce      *Nonce
 	tagKey     BoxPrecomputedSharedKey
 	position   int
+	mki        MessageKeyInfo
+}
+
+// MessageKeyInfo conveys all of the data about the keys used in this encrypted message.
```

Comment:
It might be worth pointing out that of all these fields, only the sender key is authenticated in any way, and even then only if we successfully decrypt the payload.